mlflow/pipelines/steps/split.py

```diff
@@ -238,6 +244,19 @@ def _run(self, output_directory):
         self.execution_duration = self.run_end_time - run_start_time
         return self._build_profiles_and_card(train_df, validation_df, test_df)
 
+    def _validate_user_code_output(self, pre_split_df, post_split_df):
+        import pandas as pd
+
+        if not isinstance(post_split_df, pd.DataFrame):
+            raise MlflowException(
+                message="The split data is not a DataFrame, please return the correct data"
+            )
```

Comment:
Same thing here - I think this will always be a df

Reply:
Yeah! I added this and the other function to do this validation check. So I think we can still validate the new function as well. What do you think?